### ISSUE TITLE
Tensorflow deadlock when using multi-feed_dicts or multi-queues to different gpus

### ISSUE THREAD

**Issue by ZK-Zhou** [tensorflow/tensorflow#34517]
**System information**
- pycharm
- 1.10
- Python version:3.6
- CUDA/cuDNN version:9
- GPU model and memory:GTX 1080Ti

**Describe the current behavior**
I used the data-parallel mode to train my network. For the require of research, I needed feed different category images to different GPU(all The images have same size, such as input dogs' images to GPU 0 and cats' images to GPU 1), so I initiated two tf.placeholder() and feed the different images batch to different GPU and started to train the network. It seems fine at the beginning. But after 4000 iterations or some uncertain steps, it hangs up!
**Describe the expected behavior**
 I needed feed different category images to different GPU(all The images have same size, such as input dogs' images to GPU 0 and cats' images to GPU 1) and apply the averaged gradients.
**Code to reproduce the issue**

images0, images1 = self.load_batches()

self.images0 = tf.placeholder(tf.float32, shape=(self.batch_size, self.c_dim, self.output_size, self.output_size))
self.images1 = tf.placeholder(tf.float32, shape=(self.batch_size, self.c_dim, self.output_size, self.output_size))

for  i  in range(self.config.num_gpus):
    with tf.device(device_setter):
            if i == 0:
                    self.set_tower_loss('', self.images0, Generator, Discriminator)
            else:
                    self.set_tower_loss('', self.images1, Generator, Discriminator)
   
 _, g_grads= self.sess.run([self.d_grads] , feed_dict={self.images0: images0, self.images1: images1} )

**Other info / logs**
GPU memory is employed but utility is 0 and system doe not updated.

**Comment by oanush** [comment]
@ZK-Zhou ,
Can you share a complete standalone code to reproduce the issue? also mention the tensorflow version. Thanks!

**Comment by oanush** [comment]
@ZK-Zhou ,
Any update on the issue ?Thanks!

**Comment by ZK-Zhou** [comment]
sorry to reply late! I have arranged a little model to reproduce this issue, please click my github and run it:https://github.com/ZK-Zhou/multi_queue

**Comment by ZK-Zhou** [comment]
@oanush

**Comment by ZK-Zhou** [comment]
my tensorflow version is 1.6.0

**Comment by ymodak** [comment]
Apologies for the delay in response. TF 1.6 is obsolete. Can you please test with latest version of tensorflow and check if the issue still persists? Thanks!
